replica/database.cc

```diff
@@ -929,11 +930,10 @@ future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_
     dblog.debug("Dropping {}.{}", ks_name, cf_name);
     co_await remove(*cf);
     cf->clear_views();
-    co_return co_await cf->await_pending_ops().then([this, &ks, cf, tsf = std::move(tsf), snapshot] {
-        return truncate(ks, *cf, std::move(tsf), snapshot).finally([this, cf] {
-            return cf->stop();
-        });
-    }).finally([cf] {});
+    co_await cf->await_pending_ops();
+    auto f = co_await coroutine::as_future(truncate(ks, *cf, std::move(tsf), snapshot));
+    co_await cf->stop();
+    f.get(); // re-throw exception from truncate() if any
```

Comment:
it's a shame we can't simply co_return the future... but this will work too.

Reply:
I wonder if `co_return co_await std::move(f);` will save the exception rethrowing.

Reply:
maybe, but how is that prettier :))

Reply:
it does throw, so no point in doing that.